registry/app.go

```diff
@@ -88,6 +88,8 @@ func NewApp(configuration configuration.Configuration) *App {
 }
 
 func (app *App) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Set a header with the Docker Distribution API Version for all responses.
+	w.Header().Set("Docker-Distribution-API-Version", "2.0")
```

Comment:
The only thing I would suggest, and please push back if you think this is silly, is that it should have the version resource in the header:

```
Docker-Distribution-API-Version: registry/2.0
```

Version 2.1 would be:

```
Docker-Distribution-API-Version: registry/2.1
```

Endpoint with v1 and v2, maybe in precedence order:

```
Docker-Distribution-API-Version: registry/2.0 registry/1.0
```